internal/storage/clickhouse.go

```diff
@@ -339,12 +339,33 @@ func (c *ClickHouseConnector) buildQuery(table, columns string, qf QueryFilter)
 	return query
 }
 
+func addToAddress(table, query string, toAddress string) string {
+	// This needs to move to a query param that accept multiple addresses
+	if table == "logs" {
+		if toAddress != "" {
+			query += fmt.Sprintf(" AND topic_2 = '%s'", getToAddressTopic(toAddress))
```

Comment:
Isn't topic_2 arbitrary data?
why are you comparing it to `to`?

Reply:
for logs, the second topic includes the to address, the third topic includes from. I also have a question what does the address column store in the logs table? the address in that column didn't match the to or from address in xai block explorer

Reply:
Isn't that true only for a specific kind of log?
The address column is the contract that emitted it. It's not the transaction `to` since if a contract calls another contract that emits a log, it has that contract's address